File: src/main/java/io/sch/historyscan/domain/contexts/analysis/common/Analyze.java
package io.sch.historyscan.domain.contexts.analysis.common;

import io.sch.historyscan.domain.error.HistoryScanFunctionalException;
import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureAPI;

@HexagonalArchitectureAPI
@FunctionalInterface
public interface Analyze<T> {

    T from(CodeBaseToAnalyze codeBaseToAnalyze) throws HistoryScanFunctionalException;
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/domain/contexts/analysis/common/Analyze.kt
package io.sch.historyscan.domain.contexts.analysis.common

import io.sch.historyscan.domain.error.HistoryScanFunctionalException
import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureAPI

@HexagonalArchitectureAPI
fun interface Analyze<T> {
    @Throws(HistoryScanFunctionalException::class)
    fun from(codeBaseToAnalyze: CodeBaseToAnalyze): T
}
